api/xds_protocol.rst

```diff
@@ -155,45 +231,109 @@ the resources in the :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>`
-site-specific business logic to determine the full set of resources that the client is interested in, typically based on
-the client's node identification. However, other xDS clients may specify explicit LDS/CDS resources as resource hints, for
-example if they only have a singleton listener and already know its name from some out-of-band configuration. For EDS/RDS, the
-resource hints are required.
+For other resource types, the resource hints are required. If the field is empty, that means that the client is not
+interested in any resources of the relevant type.
 
 When the resource hints are specified, the management server must supply the requested resources if they exist. The client will
 silently ignore any supplied resources that were not explicitly requested. When the client sends a new request that changes
-the *resource_names* list, the server must resend any newly requested resource, even if it previously sent it without having
-been asked for it and the resource has not changed since that time.
-
-For LDS and CDS, it is expected that the management server will provide the complete state of the LDS/CDS resources in each
-response; an absent `Listener` or `Cluster` will be deleted. For RDS or EDS, when a requested resource is
-missing, Envoy will retain the last known value for this resource except in the case where the `Cluster` or `Listener` is being
-warmed. See :ref:`Resource Warming <xds_protocol_resource_warming>` section below on the expectations during warming.
-An empty EDS/RDS :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>` is effectively a nop from the perspective of the
-respective resources in the Envoy.
-
-When a `Listener` or `Cluster` is deleted, its corresponding EDS and
-RDS resources are also deleted inside the Envoy instance. In order for
-EDS resources to be known or tracked by Envoy, there must exist an
-applied `Cluster` definition (e.g. sourced via CDS). A similar
-relationship exists between RDS and `Listeners` (e.g. sourced via
-LDS).
+the *resource_names* list, the server must resend any newly requested resources, even if it previously sent those resources
+without having been asked for them and the resources have not changed since that time.
+
+Grouping Resources into Responses
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the SotW API variants, the `Listener` and `Cluster` resource types are treated specially. The server must include the complete
+state of the world, meaning that all resources of the relevant type that are needed by the client must be included, even if they
+did not change since the last response. This means that if the server has previously sent 100 resources and only one of them has
+changed, it must resend all 100 of them, even the 99 that were not modified. However, all other resource types may be handled
+incrementally in the SotW API, meaning that resources will be grouped into responses the same way as they would be in the
```

Comment:
I think you already said this above, maybe try and find a way to avoid repeating.